util/devel/deprecationScript/replaceDeprecatedLines.py

```diff
@@ -0,0 +1,80 @@
+import sys
+import os
+
+num_replaced = 0
+
+def replaceSingleInstance(line, deprecated, replacement, include_files):
+  # Extract the file path and line number
+  # print(line)
+  file_path, line_number = line.split(':')[:2]
+  # Check if the line number is actually a number or a letter line nnn
+  # only cast to int if it is a number
+  if line_number.isdigit():
+    line_number = int(line_number)
+  else:
+    return
+  
+  
+  # Replace the $CHPL_HOME variable in the filepath
+  resolved_file_path = os.path.expandvars(file_path)
+  file_name = os.path.basename(resolved_file_path)
+  if include_files and file_name not in include_files:
+    print("Skipping ", resolved_file_path, " because it is not in the list of files to replace")
+    return
+  # print(resolved_file_path)
+  # Check if the file exists
+  # Skip if it doesn't
+  if not os.path.isfile(resolved_file_path):
+    print("File ", resolved_file_path," does not exist")
+    return
+
+  # Open the file and go to the specified line number
+  with open(resolved_file_path, 'r') as f:
+    lines = f.readlines()
+    old_line = lines[line_number - 1]
+
+  # Replace the word with another word
+  # print(old_line)
+  # print(deprecated, replacement)
+  new_line = old_line.replace(deprecated, replacement)
+  # print(new_line)
+  # Check if the line was actually replaced
+  # If it was, increment the number of replacements
+  if old_line != new_line:
+    print("Replacing ", deprecated, " with ", replacement, " in ", resolved_file_path, " at line ", line_number)
+    global num_replaced
```

Comment:
I may be wrong here, but I don't think the `global num_replaced` is required. I think its only needed if you also have a local `num_replaced`

Reply:
It didn't work without the `global num_replaced`